modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/snapshot/SnapshotManagerComponent.java

```diff
@@ -353,13 +356,27 @@ protected void beforeCreateDocument(DocumentModel doc) {
         snapDoc = session.saveDocument(snapDoc);
         snapDoc = session.move(snapDoc.getRef(), session.getParentDocumentRef(tmp.getRef()), null);
 
-        // cleanup tmp dir
+        // we need to submit the move, then wait for its completion in case of async implementation
+        commitOrRollbackAndRestartTransaction();
+        try {
+            Framework.getService(BulkService.class).await(Duration.ofMinutes(10));
```

Comment:
you need to check the return code, if false some command are still processing and you need to log an error

Reply:
I propose to throw and log. Throwing will allow to not end with an empty distrib and rollback
```suggestion
            if (!Framework.getService(BulkService.class).await(Duration.ofMinutes(10))) {
                // Interrupt the import
                throw new NuxeoException("Failed importing NXDistribution");
            }
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
            throw new NuxeoException(e);
        } finally {
            log.error("Failed importing NXDistribution");
        }
```